crates/amaru-kernel/src/memoized/transaction_output.rs

```diff
@@ -253,20 +287,9 @@ impl<'a> TryFrom<MintedTransactionOutput<'a>> for MemoizedTransactionOutput {
                     .map_err(|e| format!("invalid address: {e:?}"))?,
                 value: output.value,
                 datum: MemoizedDatum::from(output.datum_option),
-                script: output.script_ref.map(|script| match script.unwrap() {
-                    PseudoScript::NativeScript(native_script) => {
-                        PseudoScript::NativeScript(MemoizedNativeScript::from(native_script))
-                    }
-                    PseudoScript::PlutusV1Script(plutus_script) => {
-                        PseudoScript::PlutusV1Script(plutus_script)
-                    }
-                    PseudoScript::PlutusV2Script(plutus_script) => {
-                        PseudoScript::PlutusV2Script(plutus_script)
-                    }
-                    PseudoScript::PlutusV3Script(plutus_script) => {
-                        PseudoScript::PlutusV3Script(plutus_script)
-                    }
-                }),
+                script: output
+                    .script_ref
+                    .map(|script| from_memoized_script(script.unwrap())),
```

Comment:
what are we `unwrap`ing here? Why is it safe to do so?

Reply:
`unwrap` is how Pallas decided to call the method that extract the cbor serialized bytes from within a `KeepRaw` structure 😬 ... so there's nothing unsafe about this one, but it is utterly confusing I concede.